src/coreclr/jit/simdashwintrinsic.cpp

```diff
@@ -1154,50 +1188,113 @@ GenTree* Compiler::impSimdAsHWIntrinsicSpecial(NamedIntrinsic       intrinsic,
                 }
 
 #if defined(TARGET_XARCH)
-                case NI_VectorT_ConvertToInt32:
+
+                case NI_VectorT_ConvertToInt64:
+                {
+                    assert(sig->numArgs == 1);
+                    assert(simdBaseType == TYP_DOUBLE);
+                    if (IsBaselineVector512IsaSupportedOpportunistically())
+                    {
+                        NamedIntrinsic intrinsic =
+                            (simdSize == 16) ? NI_AVX512DQ_VL_ConvertToVector128Int64WithTruncation
+                                             : (simdSize == 32) ? NI_AVX512DQ_VL_ConvertToVector256Int64WithTruncation
+                                                                : NI_AVX512DQ_ConvertToVector512Int64WithTruncation;
```

Comment:
Can we push this intrinsic selection down into `gtNewSimdCvtNode` so that we don't need to duplicate it between `simdashwintrinsic.cpp` and `hwintrinsic.cpp`?

That's how most of the other `gtNewSimd*Node` APIs work, typically just passing down a different set of identifiers that allow correct selection instead (in this case, I imagine it can be selected based on the source and target CorInfoType passed down).

Reply:
Sure, we can do that. Will make these changes today.